doc/develop/test/twister/twister_blackbox.rst

```diff
@@ -0,0 +1,128 @@
+``importlib`` 's ``exec_module()`` allows us [#f2]_ to run the test as in the CLI. The result is a ``SystemExit`` ,
+which is caught by ``pytest.raises()`` (L52) and saved to ``sys_exit`` variable, so we can check
+the return code in the first ``assert`` (L67).
+
+Such a test provides us with all the outputs we typically expect of a Twister run. We can
+check the standard output or stderr thanks to ``capfd`` . The three-line block starting with
+``readouterr()`` invocation (L63-65) is commonly used in tests. stdout is written to an ``out`` variable
+and stderr to ``err`` variable. The following lines ensure that the user can still see the outputs
+in their console after running the test.
+
+Beside the standard outputs, we can also investigate the file outputs, normally placed in
+``twister-out`` directories. Most of the time, we will use the ``out_path`` fixture in conjunction
+with ``--outdir`` flag (L44) to keep test-generated files in temporary directories. In the example above,
+we read the ``testplan.json`` file to create the ``filtered_j`` variable (L55-61), which lists all testcases.
+Typical files read in blackbox tests are ``testplan.json`` and ``twister.log`` .
+
+------
+
+.. rubric:: Footnotes
+
+.. [#f1] We advise you to keep the first section of ``args`` definition intact in almost all of your tests, as it is used for the common test setup.
+         Second section is for the flags under tests, third is for any additional flags required for the test and the last section is for platforms.
+
+.. [#f2] Take note of the ``setup_class()`` class function, which allows us to run ``twister`` python file as if it were called directly (bypassing the ``__name__ == '__main__'`` check).
+
+Other functionalities
+*********************
+
+Decorators
+==========
+
+* ``@pytest.mark.usefixtures('clear_log')``
+    - allows us to use ``clear_log`` fixture from ``conftest.py`` .
+      The fixture is to become ``autouse`` in the future.
+      After that, this decorator can be removed.
+* ``@pytest.mark.parametrize('level, expected_tests', TESTDATA_X, ids=['smoke', 'acceptance'])``
+    - this is an example of ``pytest`` 's test parametrization.
+      Read up on it `here <https://docs.pytest.org/en/7.1.x/example/parametrize.html#different-options-for-test-ids>`__.
+      TESTDATAs are most often declared as class fields.
+* ``@mock.patch.object(TestPlan, 'TESTSUITE_FILENAME', testsuite_filename_mock)``
+    - this decorator allows us to use only tests defined in the ``test_data`` and
+      ignore the Zephyr testcases in the ``tests`` directory. **Note that all ``test_data``
+      tests use** ``test_data.yaml`` **as a filename, not** ``testcase.yaml`` **!**
+
+Fixtures
+========
+
+Blackbox tests use ``pytest`` 's fixtures,
+further reading on which is available
+`here <https://docs.pytest.org/en/6.2.x/fixture.html>`__.
+
+If you would like to add your own fixtures,
+consider whether they will be used in just one test file, or in many.
```

Comment:
please check line wrapping settings and ensure proper formatting of the file, as this here seems pretty aggressive :) You can have 100 characters per line.